schema/pgsql/upgrades/018.sql

```diff
@@ -0,0 +1,15 @@
+CREATE TABLE plugin (
+      type text NOT NULL,
+      name text NOT NULL,
+      version text NOT NULL,
+      author text NOT NULL,
+      config_attrs text NOT NULL,
+
+      CONSTRAINT pk_plugin PRIMARY KEY (type)
+);
+
+INSERT INTO plugin (type, name, version, author, config_attrs)
+    SELECT type, type, '0.0.0', 'Icinga GmbH', '' FROM channel;
```

Comment:
If this stays a `select`, it should at least be a `select distinct`. But I think I'd just hard code the two currently existing types here in the schema file.

Random thought: maybe we don't even want the foreign key constraint, but that question will probably only be relevant once we want to also clean up old rows from the table. But I guess we probably want to keep the rows and only flag them as inactive, it's not like this table should get huge and keeping the information sure doesn't hurt for displaying the configuration of left-over configured channels where the channel plugin is no longer present.

Reply:
Yes, the foreign key is not very helpful yet, but I had just implemented it for later use. Should I remove it for now?